js/piwik.js

```diff
@@ -5756,12 +5759,45 @@ if (typeof window.Matomo !== 'object') {
              * This should be used when your website is only available under HTTPS
              * so that all tracking cookies are always sent over secure connection.
              *
+             * Warning: If your site is available under http and https,
+             * setting this might lead to duplicate or incomplete visits.
+             *
              * @param bool
              */
             this.setSecureCookie = function (enable) {
+                if(enable && location.protocol !== 'https:') {
+                    logConsoleError("Error in setSecureCookie: You cannot use `Secure` on http.");
+                    return;
+                }
                 configCookieIsSecure = enable;
             };
 
+            /**
+             * Set the SameSite attribute for cookies to a custom value.
+             * You might want to use this if your site is running in an iframe since
+             * then it will only be able to access the cookies if SameSite is set to 'None'.
+             *
+             *
+             * Warning: 
+             * Sets CookieIsSecure to true on None, because None will only work with Secure; cookies
+             * If your site is available under http and https,
+             * using "None" might lead to duplicate or incomplete visits.
+             *
+             * @param string either Lax, None or Strict
+             */
+            this.setCookieSameSite = function (sameSite) {
+                if (sameSite === 'none' || sameSite === 'lax' || sameSite === 'strict') {
+                    sameSite = sameSite.charAt(0).toUpperCase() + sameSite.slice(1);
+                } else if (sameSite !== 'None' && sameSite !== 'Lax' && sameSite !== 'Strict') {
+                    logConsoleError('Ignored value for sameSite. Please use either Lax, None, or Strict.');
+                    return;
+                }
+                if (sameSite === 'None') {
```

Comment:
I reckon it be still great to not set `sameSite=None` on HTTP just to make sure the cookie still works on HTTP. So be great to convert it to `Lax` on HTTP protocol when it is otherwise none. I mean like 

```js
if (sameSite === 'None') {
   if (location.protocol === 'https:') {
       this.setSecureCookie(true);
   } else { sameSite = 'Lax' }
} 
```

I tested it with ` document.cookie = 'foo=bar;path=/;secure;SameSite=None'` on HTTP and it would otherwise show issues like this 
![image](https://user-images.githubusercontent.com/273120/99320004-7e76fa80-28cf-11eb-8a12-879324157149.png)

Reply:
The cookie will be set, but it will not be sent with subsequent requests, if you're in a situation where sameSite=None would be needed. Added the fallback and a warning for reverting to Lax nonetheless